### PR TITLE
Resolve VCSWP-19101

### PR DESCRIPTION
**Pull request by amahadaya**
Closes VCSWP-19101

### CODE REVIEW

File: test/Util/RequestVerifierTest.php [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+use FreeClimb\Api\Util\RequestVerifier;
+use PHPUnit\Framework\TestCase;
+
+class RequestVerifierTest extends TestCase
+{
+    protected $requestVerifierObject;
+
+    public function setUp(): void
+    {
+        $this->requestVerifierObject = new RequestVerifier();
```
Comment (ngarg-vail) [inline test/Util/RequestVerifierTest.php]: this->requestVerifier

File: test/Util/SignatureInformationTest.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+use PHPUnit\Framework\TestCase;
+
+class SignatureInformationTest extends TestCase
+{
+    protected $signatureInformationObject;
+
+    public function setUp(): void
+    {
+        $requestHeader = "t=1679944186,v1=c3957749baf61df4b1506802579cc69a74c77a1ae21447b930e5a704f9ec4120,v1=1ba18712726898fbbe48cd862dd096a709f7ad761a5bab14bda9ac24d963a6a8";
+        $this->signatureInformationObject = new SignatureInformation($requestHeader);
```
Comment (ngarg-vail) [inline test/Util/SignatureInformationTest.php]: this->signatureInformation

File: lib/Util/RequestVerifier.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+namespace FreeClimb\Api\Util;
+
+use Exception;
+use FreeClimb\Api\Util\SignatureInformation;
+
+class RequestVerifier
+{
+    public $DEFAULT_TOLERANCE = 5 * 60 * 1000;
+
+    function verifyRequestSignature(string $requestBody, string $requestHeader, string $signingSecret, int $tolerance)
```
Comment (ngarg-vail) [inline lib/Util/RequestVerifier.php]: default tolerance not used

**Review by ngarg-vail — CHANGES_REQUESTED** [review]

File: lib/Util/RequestVerifier.php (hunk 3 above)
Comment (amahadaya) [reply lib/Util/RequestVerifier.php]: Will edit this in latest commit

File: test/Util/SignatureInformationTest.php (hunk 2 above)
Comment (amahadaya) [reply test/Util/SignatureInformationTest.php]: Will edit this in latest commit

File: test/Util/RequestVerifierTest.php (hunk 1 above)
Comment (amahadaya) [reply test/Util/RequestVerifierTest.php]: Will edit this in latest commit

**Review by ngarg-vail — APPROVED** [review]

**Pull request merged**